src/routers/handlers/company/confirm.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, Response } from "express";
+import { GenericHandler } from "./../generic";
+import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
+import { Session } from "@companieshouse/node-session-handler";
+import { getCompanyProfile } from "../../../services/company/company.profile.service";
+import { buildAddress, formatForDisplay } from "../../../services/company/confirm.company.service";
+import logger from "../../../lib/Logger";
+
+export class ConfirmCompanyHandler extends GenericHandler {
+
+    constructor () {
+        super();
+        this.viewData.title = "Update a registered email address";
+    }
+
+    async get (req: Request, response: Response): Promise<Object> {
+        logger.info(`GET request to serve company confirm page`);
+        const session: Session = req.session as Session;
+        // const companyNumber = req.query.companyNumber as string;
+        const companyNumber = session.data.extra_data.companyDetails.companyNumber as string;
```

Comment:
We're going to have to cater for back link functionality too so you should wrap the 
```suggestion
        var companyNumber: string;
        if (session.data.extra_data.companyDetails.companyNumber != undefined) {
                companyNumber = session.data.extra_data.companyDetails.companyNumber;
        } else {
                companyNumber = req.query.companyNumber;
        }
```

Reply:
If session.extra_data is empty, then this will need to be populated by using the query.companyNumber so that it is available for the validation?

Reply:
If user coming from the search page it will be available in the in the session, but if they're on a page such as the `REA Update` page and click the backlink, that back link will have to supply the companyNumber